### PR TITLE
Move/rename IProcess and tests

### PR DESCRIPTION
**Pull request by duncanp-sonar**
No code changes: just file moves, namespace changes, and scope changes (moved classes and interfaces made public so they are accessible from the CFamily project).

### CODE REVIEW

File: src/Core.UnitTests/Core.UnitTests.csproj [hunk 1]
```diff
@@ -10,7 +10,7 @@
     <ProjectGuid>{11960170-E82F-4241-B1AA-552F168D11AD}</ProjectGuid>
     <RootNamespace>SonarLint.VisualStudio.Core.UnitTests</RootNamespace>
     <AssemblyName>SonarLint.VisualStudio.Core.UnitTests</AssemblyName>
-
+    <LangVersion>latest</LangVersion>
```
Comment (duncanp-sonar) [inline src/Core.UnitTests/Core.UnitTests.csproj:13]: The CFamily unit tests were using the latest language version.

File: src/Core.UnitTests/SystemAbstractions/CurrentProcessWrapperTests.cs [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+using System.Diagnostics;
+using FluentAssertions;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using SonarLint.VisualStudio.Core.SystemAbstractions;
+
+namespace SonarLint.VisualStudio.Core.UnitTests.SystemAbstractions
+{
+    [TestClass]
+    public class CurrentProcessWrapperTests
```
Comment (duncanp-sonar) [inline src/Core.UnitTests/SystemAbstractions/CurrentProcessWrapperTests.cs:29]: Test class renamed and namespace changed.

File: src/Core.UnitTests/SystemAbstractions/ProcessWrapperTests.cs [hunk 3]
```diff
@@ -18,24 +18,235 @@
...
 namespace SonarLint.VisualStudio.Core.UnitTests.SystemAbstractions
 {
     [TestClass]
     public class ProcessWrapperTests
     {
+        public TestContext TestContext { get; set; }
+
         [TestMethod]
-        public void GetCurrentProcessMainWindowHandle_ReturnsCurrent()
+        public void ProcessFactory_Start_ReturnsWrappedProcess()
```
Comment (duncanp-sonar) [inline src/Core.UnitTests/SystemAbstractions/ProcessWrapperTests.cs:40]: GitHub is getting confused here:
* the old `Core.ProcessWrapperTests` was renamed to `Core.CurrentProcessWrapperTests`
* this class was moved from `CFamily` and the namespace changed.

The code and tests haven't changed.

File: src/Core/SystemAbstractions/ICurrentProcess.cs [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+using System;
+using System.Diagnostics;
+
+namespace SonarLint.VisualStudio.Core.SystemAbstractions
+{
+    /// <summary>
+    /// Wrapper around System.Diagnostics.Process static method for testing
+    /// </summary>
+    public interface ICurrentProcess
```
Comment (duncanp-sonar) [inline src/Core/SystemAbstractions/ICurrentProcess.cs:29]: Renamed from `Core.IProcess`

File: src/Core/SystemAbstractions/IProcess.cs [hunk 5]
```diff
@@ -21,19 +21,86 @@
 using System;
 using System.Diagnostics;
 
+/// <summary>
+/// Wrappers around System.Diagnostics.Process for testing
+/// </summary>
```
Comment (duncanp-sonar) [inline src/Core/SystemAbstractions/IProcess.cs:26]: More GitHub confusion.
Moved from `CFamily` and namespace changed.
The interface and class are now public so they are visible to the CFamily assembly.

**Review by rita-sonarsource — APPROVED** [review]

**Pull request merged**